docs/advanced/Testing.md

```diff
@@ -1,8 +1,172 @@
-# Examples of Testing Sagas
+# Testing Sagas
 
-**WIP**
+**Effects return plain javascript objects**
 
-See Repository Examples:
+Those objects describe the effect and redux-saga is in charge to execute them.
+
+This makes testing very easy because all you have to do is compare that the object yielded by the saga describe the effect you want.
+ 
+## Basic Example
+
+```javascript
+console.log(put({ type: MY_CRAZY_ACTION }));
+
+/*
+{
+  @@redux-saga/IO': true,
+  PUT: {
+    channel: null,
+    action: {
+      type: 'MY_CRAZY_ACTION'
+    }
+  }
+}
+ */
+```
+
+Testing a saga that wait for a user action and dispatch
+
+```javascript
+const CHOOSE_COLOR = 'CHOOSE_COLOR';
+const CHANGE_UI = 'CHANGE_UI';
+
+const chooseColor = (color) => ({
+  type: CHOOSE_COLOR,
+  payload: {
+    color,
+  },
+});
+
+const changeUI = (color) => ({
+  type: CHANGE_UI,
+  payload: {
+    color,
+  },
+});
+
+
+function* changeColorSaga() {
+  const action = yield take(CHOOSE_COLOR);
+  yield put(changeUI(action.payload.color));
+}
+
+describe('change color saga', () => {
+  const gen = changeColorSaga();
+  it('should wait for a user to choose a color', () => {
+    expect(gen.next().value).toEqual(
+      take(CHOOSE_COLOR),
+    );
+  });
+  
+  it('should dispatch an action to change the ui', () => {
+    const color = 'red';
+    expect(gen.next(chooseColor(color)).value).toEqual(
+      put(changeUI(color))
+    );
+  });
+  
+  it('should be done', () => {
+    expect(gen.next().done).toEqual(true); 
+  });
+});
+```
+
+Another great benefit is that your tests are also your doc! They describe everything that should happen.
+
+## Branching Saga 
+
+Sometimes your saga will have different outcomes. To test the different branches without repeating all the steps that lead to it you can use the utility function **cloneableGenerator**
+```javascript
+const CHOOSE_NUMBER = 'CHOOSE_NUMBER';
+const CHANGE_UI = 'CHANGE_UI';
+const DO_STUFF = 'DO_STUFF';
+
+const chooseNumber = (number) => ({
+  type: CHOOSE_NUMBER,
+  payload: {
+    number,
+  },
+});
+
+const changeUI = (color) => ({
+  type: CHANGE_UI,
+  payload: {
+    color,
+  },
+});
+
+const doStuff = () => ({
+  type: DO_STUFF, 
+});
+
+
+function* doStuffThenChangeColor() {
+  yield put(doStuff());
+  yield put(doStuff());
+  const action = yield take(CHOOSE_NUMBER);
+  if (action.payload.number % 2 === 0) {
+    yield put(changeUI('red'));
+  } else {
+    yield put(changeUI('blue'));
+  }
+}
+
+import { put, take } from 'redux-saga/effects';
+import { cloneableGenerator } from 'redux-saga/utils';
+
+describe('doStuffThenChangeColor', () => {
```

Comment:
other parts of the docs when describing the testing story are using `tape`, consistency would be appreciated by the reading users i think